Xamarin.Forms.Core/RadioButton.cs

```diff
@@ -0,0 +1,425 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Windows.Input;
+using Xamarin.Forms.Internals;
+using Xamarin.Forms.Platform;
+
+namespace Xamarin.Forms
+{
+	[RenderWith(typeof(_RadioButtonRenderer))]
+	public partial class RadioButton : IToggleButtonElement, IToggleButtonController, ITextButtonElement
+	{
+		static Dictionary<string, List<WeakReference<RadioButton>>> _groupNameToElements;
```

Comment:
couldn't you use a `ConditionalWeakTable` ?https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.conditionalweaktable-2?view=netframework-4.7.1

Reply:
@StephaneDelcroix 
Thanks for bringing the idea of `ConditionalWeakTable`. How do you see using it instead of the `_groupNameToElements` dictionary and how would it better?  I need to get all the radio buttons which are part of the same group (a string).

Reply:
the problem with this approach is the List keeps getting populated, and so the performance degrades over time

Reply:
I've seen `PurgeDeads`, but we probably can do better